homeassistant/components/egps/switch.py

```diff
@@ -0,0 +1,82 @@
+"""Switch implementation for EGPS Platform."""
+
+from typing import Any
+
+from pyegps import __version__ as PYEGPS_VERSION
+from pyegps.exceptions import EgpsException
+from pyegps.powerstrip import PowerStrip
+
+from homeassistant.components.switch import SwitchDeviceClass, SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import STATE_OFF, STATE_ON
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add EGPS sockets for passed config_entry in HA."""
+    powerstrip: PowerStrip = hass.data[DOMAIN][config_entry.entry_id]
+
+    async_add_entities(
+        EGPowerStripSocket(powerstrip, socket)
+        for socket in range(powerstrip.numberOfSockets)
```

Comment:
I think I suggested to get rid of the brackets, but should we update the entites before adding them?
```suggestion
      [
          EGPowerStripSocket(powerstrip, socket)
          for socket in range(powerstrip.numberOfSockets)
     ],
    update_before_add=True,
```